trace_decoder/src/core.rs

```diff
@@ -157,7 +162,10 @@ pub fn entrypoint(
                     checkpoint_consolidated_hash,
                     contract_code: contract_code
                         .into_iter()
-                        .map(|it| (keccak_hash::keccak(&it), it))
+                        .map(|it| match &world {
+                            Either::Left(_type1) => (KeccakHash::hash(&it), it),
+                            Either::Right(_type2) => (PoseidonHash::hash(&it), it),
```

Comment:
Nit (to avoid hard coding a second time):
```suggestion
                            Either::Left(_type1) => (Type1World::CodeHasher::hash(&it), it),
                            Either::Right(_type2) => (Type2World::CodeHasher::hash(&it), it),
```